### ISSUE TITLE
Webpack isn't producing valid sourcemaps?

### ISSUE THREAD

**Issue by kenhoff** [webpack/webpack#7616]
# tl;dr

- According to https://sourcemaps.io/ and [`sourcemap-validator`](https://www.npmjs.com/package/sourcemap-validator), the sourcemaps produced by webpack are invalid
- using `devtool: "source-map"`, a barebones webpack config, and `Version: webpack 4.12.2`
- sample repro repo available at https://github.com/kenhoff/webpack-sourcemap-testing

(this might be an issue with `sourcemap-validator`, but the error that it's throwing (in the "current behavior" section below) seems pretty legit)

# Bug report

**What is the current behavior?**

Running webpack with the following config:

```
const webpack = require('webpack');
const path = require('path');

module.exports = {
    entry: './src/index.js',
    output: {
        path: path.resolve(__dirname, 'dist'),
        filename: 'bundle.js'
    },
    devtool: 'source-map'
};
```

- produces `dist/bundle.js` and `dist/bundle.js.map`.
- those files are able to be used in Chrome dev tools with source mappings 👍 
- but running those files through `sourcemap-validator` (or https://sourcemaps.io/, and consequently Sentry) produces the following error: 

```
/Users/kenhoff/node_modules/sourcemap-validator/index.js:146
          throw new Error(errMsg);
          ^

Error: Warning: mismatched names
Expected: installedModules || 'installedModules' || 'installedModules' || "installedModules" || "installedModules"
Got:    var installedM ||       var installedMod ||     var installedMod ||     var installedMod ||     var installedMod
Original Line:          var installedModules = {};
Mapping: 1:17→2:0 "installedModules" in webpack:///webpack/bootstrap
    at /Users/kenhoff/node_modules/sourcemap-validator/index.js:146:17
    at Array.forEach (<anonymous>)
    at SourceMapConsumer_eachMapping [as eachMapping] (/Users/kenhoff/node_modules/sourcemap-validator/node_modules/source-map/lib/source-map/source-map-consumer.js:570:10)
    at validate (/Users/kenhoff/node_modules/sourcemap-validator/index.js:83:12)
    at Object.<anonymous> (/Users/kenhoff/dev/webpack-sourcemap-testing/test-sourcemaps.js:6:1)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
``` 

**If the current behavior is a bug, please provide the steps to reproduce.**

Repro repo here: https://github.com/kenhoff/webpack-sourcemap-testing

Repro instructions for that repo are on the readme. 

If starting from scratch, use the following webpack config:

```
const webpack = require('webpack');
const path = require('path');

module.exports = {
    entry: './src/index.js',
    output: {
        path: path.resolve(__dirname, 'dist'),
        filename: 'bundle.js'
    },
    devtool: 'source-map'
};
```

and use `sourcemap-validator` to test `dist/bundle.js` for validity.

**What is the expected behavior?**

Expected behavior is for `sourcemap-validator` to read the sourcemap as valid (e.g., not throw an error).

(It looks like it might have something to do with some misaligned whitespace, but I'm really not sure :\ )

**Other relevant information:**
webpack version: `Version: webpack 4.12.2`
Node.js version: `v9.4.0`
Operating System: macOS `10.13.4`
Additional tools: none

**Comment by sokra** [comment]
Ok two changes are necessary to achieve better source maps that validate correctly:

https://github.com/webpack/webpack-sources/blob/b643bfba58c5820f9cf8f00a19979c1208e47692/lib/OriginalSource.js#L12-L38

change this to:

``` js
function _splitCode(code) {
	return code.split(/([\w$_0-9]+)/);
}
```

Maybe even improve the regexp a bit to match `IdentifierName` from the EcmaScript Spec: https://www.ecma-international.org/ecma-262/5.1/#sec-7.6

and

https://github.com/webpack-contrib/uglifyjs-webpack-plugin/blob/16d4eb345905a284bf1d33be66cb4a9572d91c2c/src/uglify/minify.js#L136-L138

change this to:

``` js
uglifyOptions.sourceMap = {
  content: inputSourceMap,
  includeSources: true
};
```

Please send PRs 😄

**Comment by sokra** [comment]
cc @evilebottnawi

**Comment by evilebottnawi** [comment]
@sokra should we enable this by default? or better add to docs?

**Comment by sokra** [comment]
@evilebottnawi this can't be specified via options, can it?

**Comment by evilebottnawi** [comment]
@sokra we can modify `sourceMap: true` -> `sourceMap: { content: inputSourceMap, includeSources: true }`

**Comment by sokra** [comment]
ah I tried again and the uglifyjs-webpack-plugin change is actually not needed at all.

Sorry for bothering...

**Comment by sokra** [comment]
the webpack-sources change very likely affects performance very negatively.

**Comment by zp1996** [comment]
I aslo with this bug, webpack version is 4.17.1, error trace line number it not right

**Comment by Hoten** [comment]
@kenhoff will you be sending in a PR with the changes @sokra proposed? I can take over if you like.

**Comment by kenhoff** [comment]
@Hoten go for it, all you! 👍

**Comment by Hoten** [comment]
On a related note, why is the bootstrap code prefixed with some whitespace?

https://github.com/webpack/webpack/blob/aecdeb35a662953285603ca09a004ee752d2fb60/lib/MainTemplate.js#L414

**Comment by adaniliuk** [comment]
@kenhoff did the PR https://github.com/webpack/webpack-sources/pull/41 fix the original issue? webpack@4.20.0 uses webpack-sources@1.3.0, but the issue is still reproducible for https://github.com/kenhoff/webpack-sourcemap-testing.

**Comment by adaniliuk** [comment]
@sokra @Hoten shouldn't this issue be reopened because https://github.com/webpack/webpack-sources/pull/41 didn't really fix it? 

Just in case I've forked @kenhoff repo and updated webpack version to the latest https://github.com/adaniliuk/webpack-sourcemap-testing

There is the output:
```
$ webpack
Hash: 9bd50b65233535f2f320
Version: webpack 4.22.0
Time: 361ms
Built at: 2018-10-22 10:42:17
        Asset       Size  Chunks             Chunk Names
    bundle.js  984 bytes       0  [emitted]  main
bundle.js.map   4.52 KiB       0  [emitted]  main
Entrypoint main = bundle.js bundle.js.map
[0] ./src/index.js 46 bytes {0} [built]

WARNING in configuration
The 'mode' option has not been set, webpack will fallback to 'production' for this value. Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/concepts/mode/

$ node test-sourcemaps.js
/webpack-sourcemap-testing/node_modules/sourcemap-validator/index.js:146
          throw new Error(errMsg);
          ^
Error: Warning: mismatched names
Expected: installedModules || 'installedModules' || 'installedModules' || "installedModules" || "installedModules"
Got:  	var installedM ||  	var installedMod ||  	var installedMod ||  	var installedMod ||  	var installedMod
Original Line:  	var installedModules = {};
Mapping: 1:17→2:0 "installedModules" in webpack:///webpack/bootstrap
    at /webpack-sourcemap-testing/node_modules/sourcemap-validator/index.js:146:17
    at Array.forEach (<anonymous>)
    at SourceMapConsumer_eachMapping [as eachMapping] (/webpack-sourcemap-testing/node_modules/sourcemap-validator/node_modules/source-map/lib/source-map/source-map-consumer.js:570:10)
    at validate (/webpack-sourcemap-testing/node_modules/sourcemap-validator/index.js:83:12)
    at Object.<anonymous> (/webpack-sourcemap-testing/test-sourcemaps.js:6:1)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
```

**Comment by adaniliuk** [comment]
@kenhoff was this issue really solved to you?

**Comment by kenhoff** [comment]
Hi friends - sorry about the delay on this. I ran through my sample repo (https://github.com/kenhoff/webpack-sourcemap-testing) and updated to the latest versions of everything. Appears to still be an issue. Looks like there's a newer version of the issue here ----> #8302
